docs/cub/api_docs/device_wide.rst

```diff
@@ -11,6 +11,47 @@ Device-Wide Primitives
    ../api/device
 
 
+Determining Temporary Storage Requirements
+++++++++++++++++++++++++++++++++++++++++++++++++++
+
+**Two-Phase API** (Traditional)
+
+Most CUB device-wide algorithms follow a two-phase usage pattern:
+
+1. **Query Phase**: Call the algorithm with ``d_temp_storage = nullptr`` to determine required temporary storage size
+2. **Execution Phase**: Allocate storage and call the algorithm again to perform the actual operation
+
+**What arguments are needed during the query phase?**
+
+* **Required**: Data types (via template parameters and iterator types) and problem size (``num_items``)
+* **Can be nullptr/uninitialized**: All input/output pointers (``d_in``, ``d_out``, etc.)
+* **Note**: The algorithm does not access input data during the query phase
+
+Example pattern:
+
+.. code-block:: c++
```

Comment:
we stopped using `code-block` and rather use `literalinclude`s now linked to standalone .cu source examples.